### PR TITLE
cr-config: calculating the hash-id if needed

### PR DESCRIPTION
**Pull request by nir-sopher**
Hi,
When adding a server via the old traffic ops, the xmpp_id field is saved to be the host_name.
This field is used later on as an hashId, as part of cr-config,

However, when using the API (and therefore portal v2) this field is not created, causing the hashId to be empty and interfere with content affinity.

The suggested change comes to fix the issues for existing caches:
- Caches added by the old UI: no change in hashId
- Caches  added by the portal: hashId is calculated in the same way as by the old OPs, but now during the cr-config creation

I believe that as this is a "calculated" field and not provided by the user, the approach of calculating it each time is better then fixing it in the DB, as it allows future changes in the algorithem.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/991/
Test PASSed.

**Comment by mitchell852** [comment]
@elsloo - whatcha think?

**Comment by elsloo** [comment]
Looks good to me.

File: traffic_ops/app/lib/UI/Topology.pm [hunk 1]
```diff
@@ -264,7 +264,7 @@ sub gen_crconfig_json {
             $data_obj->{'contentServers'}->{ $row->host_name }->{'ip6'}           = ( $row->ip6_address || "" );
             $data_obj->{'contentServers'}->{ $row->host_name }->{'profile'}       = $row->profile->name;
             $data_obj->{'contentServers'}->{ $row->host_name }->{'type'}          = $row->type->name;
-            $data_obj->{'contentServers'}->{ $row->host_name }->{'hashId'}        = $row->xmpp_id;
+            $data_obj->{'contentServers'}->{ $row->host_name }->{'hashId'}        = $row->xmpp_id ? $row->xmpp_id : $row->host_name;
```
Comment (mitchell852) [inline traffic_ops/app/lib/UI/Topology.pm]: would this allow an xmpp_id = "" which would probably not be a good thing. do we need a better test like isDefined and neq empty string?

**Comment by nir-sopher** [comment]
As far as I tested, empty string is "false".
Nir
